doc/specs/#1595 - Suggestions UI/Suggestions-UI.md

```diff
@@ -0,0 +1,679 @@
+
+#### Description Tooltips
+
+> **Note**: _This is left as a future consideration for the initial draft of
+> this spec. I'd like to flesh out [shell-driven autocompletion] more before
+> committing any plans here._
+
+It would be beneficial for the Suggestions UI to display additional context to
+the user. Consider a extension that provides some commands for the user, like a
+hypothetical "Docker" extension. The extension author might be able to give the
+commands simplified names, but also want to expose a more detailed description
+of the commands to the user.
+
+Or consider the Suggestions UI when invoked by [shell-driven autocompletion].
+The shell might want to provide help text to the user with each of the
+suggestions. This would allow a user to browse through the suggestions that they
+might not know about, and learn how they work before committing to one.
+
+Only the help text for the currently hovered command should be presented to the
+user. To support this kind of UX, we'll add an optional flyout of some sort to
+display with the Suggestions UI. This flyout will only appear if there's more
+information provided to the Terminal.
+
+This might be in the form of a `TeachingTip`, as in this example:
+
+![](tasks-suggestions.gif)
+
+Actions in the settings could also accept an optional `description` property, to
+specify the string that would be presented in that flyout.
+
+#### Pre-filtering the UI & filter by source
+
+> **Note**: _This is a brainstorm I considered while writing this spec. I would
+> not not include it in the v1 of this spec. Rather, I'd like to leave it for
+> where we might go with this UX in the future._
+
+Do want to support different _types_ of nesting? So instead of just the default,
+there could be something like `nesting: "source"`, to create a menu structured
+like:
+
+```
+Suggestions UI
+├─ Recent Commands...
+│  ├─ git checkout main
+│  ├─ git fetch
+│  └─ git pull
+├─ Recent Directories...
+│  ├─ d:\dev
+│  ├─ d:\dev\public
+│  └─ d:\dev\public\terminal
+├─ Saved tasks...
+│  ├─ Git...
+│  │  └─ git commit -m "
+│  │  └─ git log...
+│  └─ bx & runut
+└─ Docker
+   ├─ docker build --platform linux/amd64 <path>
+   └─ docker logs -f --tail <lines_count> <container_name>
+```
+
+> **Note**
+> I'm using `Docker` as an example fragment extension that provides
+> some `docker` commands. When grouping by `"source"`, we could pull those into
+> a separate top-level entry. When not grouping by `"source"`, those would still
+> show up with the rest of `tasks`. )
+
+#### Store recent commands across sessions
+
+> **Note**
+> _I'm not sure we really want to put this in this spec or not, hence
+> why it is in the "Future considerations" section. I think it is worth
+> mentioning. This might be better served in the [shell integration] doc._
+
+We'll probably want a way for recent commands to be saved across sessions. That way, your `cmd.exe` command history could persist across sessions. We'd need:

```

Comment:
(thinking out loud)
This could be useful for things like portable mode, ssh sessions, and cross-machine scenarios. I could be convinced.
On the other hand, this is also a bit dangerous in that you might get a command that straight up won't work in any of the scenarios mentioned above. And _that_ I don't like.